From d624076f62f0edeb5397b99c1b5d88be0b3c86c9 Mon Sep 17 00:00:00 2001
From: Johnny Mnemonic <jm@machine-hall.org>
Date: Mon, 27 Jan 2025 15:43:18 +0100
Subject: [PATCH 54/63] Partially revert "x86/efistub: Drop long obsolete UGA
 support"

ia64 still has an implementation for efi_systab_show_arch(), so although
no issues were observed on an rx2620 without this partial revert, it could
still be needed for some other machine or boot loader.

This partially reverts commit 144d52dd8fc83a082a275e1b663e7454d2b616a4.
---
 drivers/firmware/efi/efi.c | 3 +++
 include/linux/efi.h        | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/drivers/firmware/efi/efi.c b/drivers/firmware/efi/efi.c
index c7c32ba6dd75..2cb06c4d55eb 100644
--- a/drivers/firmware/efi/efi.c
+++ b/drivers/firmware/efi/efi.c
@@ -148,6 +148,9 @@ static ssize_t systab_show(struct kobject *kobj,
 	if (efi.smbios != EFI_INVALID_TABLE_ADDR)
 		str += sprintf(str, "SMBIOS=0x%lx\n", efi.smbios);
 
+	if (IS_ENABLED(CONFIG_IA64))
+		str = efi_systab_show_arch(str);
+
 	return str - buf;
 }
 
diff --git a/include/linux/efi.h b/include/linux/efi.h
index 9e111e5efc8e..7bfcb965acef 100644
--- a/include/linux/efi.h
+++ b/include/linux/efi.h
@@ -1294,6 +1294,8 @@ struct linux_efi_memreserve {
 
 void __init efi_arch_mem_reserve(phys_addr_t addr, u64 size);
 
+char *efi_systab_show_arch(char *str);
+
 /*
  * The LINUX_EFI_MOK_VARIABLE_TABLE_GUID config table can be provided
  * to the kernel by an EFI boot loader. The table contains a packed
-- 
2.25.1

